distributed/security.py

```diff
@@ -149,10 +167,36 @@ def temporary(cls, **kwargs):
 
     def _set_field(self, kwargs, field, config_name):
         if field in kwargs:
-            out = kwargs[field]
+            val = kwargs[field]
+        else:
+            val = dask.config.get(config_name)
+        setattr(self, field, val)
+
+    def _set_tls_version_field(self, kwargs, field, config_name, default=None):
+        if field in kwargs:
+            val = kwargs[field]
+            valid = {None, ssl.TLSVersion.TLSv1_2, ssl.TLSVersion.TLSv1_3}
+            if val not in valid:
+                raise ValueError(
+                    f"{field}={val!r} is not supported, expected one of {valid}"
+                )
+            if val is None:
+                val = default
         else:
-            out = dask.config.get(config_name)
-        setattr(self, field, out)
+            val = dask.config.get(config_name)
+            valid = {
+                1.2: ssl.TLSVersion.TLSv1_2,
+                1.3: ssl.TLSVersion.TLSv1_3,
+                None: default,
+            }
```

Comment:
Also a nit: Can we easily move the definition of `valid` to be before this `else` block (and the above `if` block) so that there's only one place where we specify valid tls versions? Right now it appears we're repeating some logic in the the `if` and `else` blocks

Reply:
The values accepted differ slightly between config file and code - it seems easier to slightly duplicate these dicts. FWIW, TLS versions rarely release, I don't expect this to be a maintenance burden.